src/runtime_src/xdp/profile/database/static_info/device_info.cpp

```diff
@@ -318,13 +319,13 @@ namespace xdp {
       return ;
 
     switch (moduleType) {
-    case 0:
+    case static_cast<int>(module_type::core):
```

Comment:
Why do we need the static_cast?

Reply:
This is because type safety provided by `enum class` over traditional enum. I have changed enum definition to traditional type for implicit conversion of enumerators.